cla_backend/apps/knowledgebase/tests/test_csv_2_fixture.py

```diff
@@ -0,0 +1,1443 @@
+        return min_range["pk"], max_range["pk"]
+
+    def sort_article_category_matrices(self, output_category_matrix, expected_category_matrix):
+        output_sorted_acm = sorted(output_category_matrix, key=lambda x: x["fields"]["article_category"])
+        expected_sorted_acm = sorted(expected_category_matrix, key=lambda x: x["fields"]["article_category"])
+        return output_sorted_acm, expected_sorted_acm
+
+    def slice_records(self, output_records, expected_records, startRange=None, endRange=None):
+        output_sliced = output_records[startRange:endRange]
+        expected_sliced = expected_records[startRange:endRange]
+        return output_sliced, expected_sliced
+
+    def compare_dicts(self, outputDict, expectedDict):
+        for key, value in expectedDict.items():
+            self.assertIsInstance(value, type(outputDict[key]))
+            if isinstance(value, int):
+                self.assertEqual(outputDict[key], value)
+            elif isinstance(parse_datetime(value), datetime):
+                self.compare_datetimes(parse_datetime(outputDict[key]), parse_datetime(value))
+            else:
+                self.assertEqual(outputDict[key], value)
+
+    def compare_datetimes(self, outputDatetime, expectedDatetime):
+        time_diff = outputDatetime - expectedDatetime
+        time_diff_in_seconds = time_diff.total_seconds()
+        self.assertLessEqual(time_diff_in_seconds, 0.5)
+
+    def compare_list_of_records(self, outputList, expectedList):
+        for expectedObject, outputObject in zip(expectedList, outputList):
+            self.compare_length(outputObject, expectedObject)
+            for key, value in expectedObject.items():
+                self.assertIsInstance(value, type(outputObject[key]))
+                if isinstance(value, dict):
+                    self.compare_dicts(outputObject[key], value)
+                else:
+                    self.assertEqual(outputObject[key], value)
+
+    def compare_length(self, output, expected):
+        self.assertAlmostEqual(len(output), len(expected))
+
+    def compare_article_category_matrices(self, output, expected, pk_range):
+        for outputObj, expectedObj in zip(output, expected):
+            self.compare_length(output, expected)
+            for key, value in expectedObj.items():
+                if key == "pk":
+                    self.assertIsInstance(value, int)
+                    self.assertIsInstance(value, type(outputObj[key]))
+                    min_pk_value, max_pk_value = pk_range
+                    self.assertGreaterEqual(value, min_pk_value)
+                    self.assertLessEqual(value, max_pk_value)
+                elif isinstance(value, dict):
+                    self.compare_dicts(outputObj[key], value)
+                else:
+                    self.assertIsInstance(value, type(outputObj[key]))
+                    self.assertEqual(outputObj[key], value)
```

Comment:
All of these lines would be easier to read if we assigned `outputObj[key]` to a variable as soon as we enter this `for` loop (like we did on the previous ticket)

Reply:
Done